commcare_connect/opportunity/models.py

```diff
@@ -282,6 +290,45 @@ class CompletedWork(models.Model):
     entity_name = models.CharField(max_length=255, null=True, blank=True)
     reason = models.CharField(max_length=300, null=True, blank=True)
 
+    # TODO: add caching on this property
+    @property
+    def completed_count(self):
+        """Returns the no of completion of this work. Includes duplicate submissions."""
+        visits = self.uservisit_set.filter(status=VisitValidationStatus.approved).values_list(
+            "deliver_unit_id", flat=True
+        )
+        unit_counts = Counter(visits)
+        required_deliver_units = self.payment_unit.deliver_units.filter(optional=False).values_list("id", flat=True)
+        optional_deliver_units = self.payment_unit.deliver_units.filter(optional=True).values_list("id", flat=True)
```

Comment:
same nit about making this one query.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/289/commits/8664b8b2bfbfd8a94a1cf65e0da21bcc39965968